android/quest/src/main/java/org/smartregister/fhircore/quest/ui/appsetting/AppSettingViewModel.kt

```diff
@@ -93,20 +95,45 @@ constructor(
     if (!appId.isNullOrEmpty()) {
       when {
         hasDebugSuffix() -> loadConfigurations(context)
-        else -> fetchRemoteConfigurations(appId, context)
+        else -> fetchRemoteConfigurations(appId, QuestBuildConfig.VERSION_NAME.substringBefore("-"), context)
       }
     }
   }
 
-  private fun fetchRemoteConfigurations(appId: String?, context: Context) {
+  private fun fetchRemoteConfigurations(appId: String?, appVersion: String?, context: Context) {
     viewModelScope.launch {
       try {
         showProgressBar.postValue(true)
-        Timber.i("Fetching configs for app $appId")
-        val compositionResource =
-          withContext(dispatcherProvider.io()) {
-            configurationRegistry.fetchRemoteComposition(appId)
-          }
+
+        Timber.i("Fetching configs for app $appId version $appVersion")
+
+        var compositionResource: Composition? = null
+        try {
+          val implementationGuideResource =
+            configurationRegistry.fetchRemoteImplementationGuide(appId, appVersion)
+
+          compositionResource =
+            if (implementationGuideResource != null) {
+              configurationRegistry.addOrUpdate(implementationGuideResource)
+
+              val compositionReference =
+                implementationGuideResource.definition.resource[0].reference.reference
+
+              val compositionIdWithHistory = compositionReference?.substringAfter('/')
+              val compositionId = compositionIdWithHistory?.substringBefore('/')
+              val compositionVersion = compositionIdWithHistory?.substringAfterLast('/', "")
+
+              withContext(dispatcherProvider.io()) {
+                configurationRegistry.fetchRemoteCompositionById(compositionId, compositionVersion)
+              }
```

Comment:
This is definitely not the goal here. The IG should be self contained, i.e. the composition itself must be embedded in the IG, it should not contain a remove reference to the composition.

Reply:
Does this relate embedding the composition in the IG comment above? See my question in the comment https://github.com/opensrp/fhircore/pull/3189/files#r1570397539

Reply:
Yea it's the same, your point makes sense, I misunderstood this.